pallets/nfts/src/tests.rs

```diff
@@ -1828,6 +1953,89 @@ fn approval_lifecycle_works() {
 	});
 }
 
+#[test]
+fn check_approval_works() {
+	new_test_ext().execute_with(|| {
+		let collection_id = 0;
+		let item_id = 42;
+		let collection_owner = account(1);
+		let item_owner = account(2);
+		let delegate = account(3);
+
+		Balances::make_free_balance_be(&item_owner, 100);
+		assert_ok!(Nfts::force_create(
+			RuntimeOrigin::root(),
+			collection_owner.clone(),
+			default_collection_config()
+		));
+		assert_ok!(Nfts::force_mint(
+			RuntimeOrigin::signed(collection_owner.clone()),
+			collection_id,
+			item_id,
+			item_owner.clone(),
+			default_item_config()
+		));
+
+		let deadline = 10;
+		let error = Err(Error::<Test>::ApprovalExpired.into());
+		for case in Vec::<(
+			// Deadline of the collection and item approval.
+			(Option<BlockNumberFor<Test>>, Option<BlockNumberFor<Test>>),
+			// Check results when collection or item approval deadline has passed.
+			(DispatchResult, DispatchResult),
+			// Check results when both collection and item approval deadline has passed.
+			(DispatchResult, DispatchResult),
+		)>::from(vec![
+			// No deadline set for both collection and item approval.
+			((None, None), (Ok(()), Ok(())), (Ok(()), Ok(()))),
+			// Set deadline for both. Collection approval expires first.
+			((Some(deadline), Some(deadline + 1)), (error, Ok(())), (error, error)),
+			// Set deadline for both. Item approval expires first.
+			((Some(deadline + 1), Some(deadline)), (Ok(()), Ok(())), (error, error)),
+		]) {
```

Comment:
I prefer doing it this way because there are two different cases for the deadline. And you did suggest creating a for loop here: https://github.com/r0gue-io/pop-node/pull/387#discussion_r1871023122

Reply:
I don't, the test is a puzzle to understand. Please refactor.